### DISCUSSION TITLE
Please return QuectelStatus, QuectelSendSms and so on !

### DISCUSSION THREAD

**Discussion by CpServiceSpb** [RoEdAl/asterisk-chan-quectel#5] [General]
I founded your fork, your version of Asterisk Quectel channel driver and saw more options then at [IchthysMaranatha](https://github.com/IchthysMaranatha/asterisk-chan-quectel).
I successfully built your fork on Ubuntu 22.04 x64 (Intel) LTS for Asterisk 20.4.0.
I loaded it.
I use some macros in my dial plans where for example QuectelStatus is used.
But I got the error - "No application 'QuectelStatus' for extension ..."

I investigated more deep and found out that these application absent.
And there are not app.c/app.h and manager.c/manager.h files at all.

Please return complete all applications which are at [IchthysMaranatha](https://github.com/IchthysMaranatha/asterisk-chan-quectel) fork looking like as at Dongle channel also.

**Comment by RoEdAl** [comment]
Can you show me an example of `QuectelStatus` usage in your dialplan? For me `QuectelStatus` function seems to be a little hackish.

**Reply by CpServiceSpb** [reply]
There is not only QuectelStatus, there are QuectelSendUSSD and so on.
Here us part of my dialplan:
```
exten => s,1,QuectelStatus(${ARG1},Quectel_Status)
exten => s,n(line3),ExecIf($["${Quectel_Status}" = "2"]?Goto(line21))
exten => s,n(line4),ExecIf($["${Quectel_Status}" = "-1"]?Playback(nnot-installed):Goto(line6))
exten => s,n(line5),Goto(hangup)
exten => s,n(line6),ExecIf($["${Quectel_Status}" = "1"]?Playback(net-connected):Goto(line8))
exten => s,n(line7),Goto(hangup)
exten => s,n(line8),ExecIf($["${Quectel_Status}" = "3"]?:Goto(line12))
exten => s,n,ExecIf($[${ARG7}=1]?Playback(busy))
exten => s,n,ExecIf($[${ARG7}=2]?Playback(by-not-accessible))
exten => s,n,Goto(hangup)
exten => s,n(line12),ExecIf($["${Quectel_Status}"] = "2"?Goto(line21))
exten => s,n,ExecIf($["${Quectel_Status}" = "-1"]?:Goto(line14))
exten => s,n,Goto(hangup)
exten => s,n(line15),ExecIf($["${Quectel_Status}" = "1"]?:Goto(line17))
exten => s,n,Goto(hangup)
exten => s,n(line17),ExecIf($["${Quectel_Status}" = "3"]?:Goto(line21))
exten => s,n,ExecIf($[${ARG7}=1]?Playback(busy))
exten => s,n,ExecIf($[${ARG7}=2]?Playback(by-not-accessible))
exten => s,n,Goto(hangup)
```

Why this function is seemed hackish ?

**Reply by RoEdAl** [reply]
This was my mistake. I thought that `QuectelStatus` function is used to force device status. Reading device status is OK. 

For me instead of:

```
exten => s,1,QuectelStatus(${ARG1},Quectel_Status)
```

a more convenient solution would be:

```
exten => s,1,Set(Quectel_Status=${QuectelStatus(${ARG1})})
```

As of SMS and USSD you may try:

```
exten => s,1,SHELL(asterisk -x 'quectel sms send <device> <number> <msg>')
```

**Reply by CpServiceSpb** [reply]
> exten => s,1,Set(Quectel_Status=${QuectelStatus(${ARG1})})
But QuectelStatus is not implemented in your forl.

> exten => s,1,SHELL(asterisk -x 'quectel sms send <device> <number> <msg>')
No, no. Why is SHELL for ?

Why did you cut all these functionality ?

**Reply by RoEdAl** [reply]
[Now](https://github.com/RoEdAl/asterisk-chan-quectel/commit/1e9430c0c0bcfbdf5c4c2f79388ece42575d0524) you may try `QUECTEL_STATUS` and `QUECTEL_STATUS_EX` dialplan **functions**:

```
exten => s,1,Set(QSTATUS=${QUECTEL_STATUS(${ARG1})})
exten => s,2,Set(QQSTATUS=${QUECTEL_STATUS_EX(${ARG1})})
```

Also `QuectelSendSms` and `QuectelSendUssd` functions are renamed to `QUECTEL_SEND_SMS` and `QUECTEL_SEND_USSD` ones respectively.

**Reply by CpServiceSpb** [reply]
Why didn' t you remain original functions name ?

**Reply by RoEdAl** [reply]
Why should I keep the original function names?

If you want original function names just use great and stable [IchthysMaranatha's](https://github.com/IchthysMaranatha/asterisk-chan-quectel) fork.

**Reply by CpServiceSpb** [reply]
I thought yours is more stable.
